### PR TITLE
Reduce closure allocation

### PR DESCRIPTION
**Pull request by benaadams**
Removes a couple closure allocations identified in https://github.com/aspnet/KestrelHttpServer/issues/286

### CODE REVIEW

**Comment by benaadams** [comment]
Still 15 closure captures; but size and number is reduced

**Comment by benaadams** [comment]
2002 requests before 

img

2004 requests after

img

**Comment by benaadams** [comment]
Needs to go further; some of the closures remaining are too heavy...

**Comment by benaadams** [comment]
Next iteration for 2005 requests

allocs

Need to clean up names before I add commit

File: test/Microsoft.AspNet.Server.KestrelTests/Program.cs [hunk 1]
```diff
@@ -27,7 +28,7 @@ public class Program
 
         public int Main()
         {
-            return new Xunit.Runner.Dnx.Program(env, sp, _libraryManager).Main(new string[]
+            return new Xunit.Runner.Dnx.Program(env, sp, _libraryManager, _shutdown).Main(new string[]
```
Comment (davidfowl) [inline test/Microsoft.AspNet.Server.KestrelTests/Program.cs]: Revert that

**Comment by benaadams** [comment]
No sure about this.

**Pull request closed without merging**